phase3/all.jsonnet

```diff
@@ -1,5 +1,6 @@
 function(cfg)
   local if_enabled(addon, manifest) = if std.objectHas(cfg.phase3, addon) && cfg.phase3[addon] then manifest else {};
+  local if_cni_plugin(plugin, manifest) = if std.objectHas(cfg.phase3, "cni") && cfg.phase3["cni"] == plugin then manifest else {};
```

Comment:
What's the experience if someone accidentally misspells the name of the CNI provider (or specifies `weave-net` instead of `weave`, or chooses a provider we don't actually support)? Do we silently treat it as if they disabled CNI? Is it possible to give the user an error message instead?

Reply:
I'm not sure it's possible using the current jsonnet usage, but maybe we create a new `validate` script in the phase3 dir or add a validation function to `gen`. What do you think?